DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/Purity.scala

```diff
@@ -214,10 +217,27 @@ object Purity {
 
         val declaredMethods = project.get(DeclaredMethodsKey)
 
-        val projMethods: Seq[DefinedMethod] =
+        val allMethods: Traversable[DefinedMethod] =
             for (cf ← project.allProjectClassFiles; m ← cf.methodsWithBody)
                 yield declaredMethods(m)
 
+        val projMethods = allMethods.filter { m ⇒
+            val pn = m.definedMethod.classFile.thisType.packageName
+            packages match {
+                case None ⇒
+                    isJDK || !pn.startsWith("java/") && !pn.startsWith("javax") &&
```

Comment:
val jdkPackages = List("java")
```suggestion
                   private val JDKPackages = Array(
        "java","javax","javafx","jdk","sun","oracle","com/sun","com/oracle","netscape",
        "org/ietf/jgss","org/jcp/xml/dsig/internal","com/omg","com/w3c/dom","com/xml/sax")
        
                    isJDK || jdkPackages.forall(!pn.startsWith(_))
```